lib/cdo/sauce_connect.rb

```diff
@@ -0,0 +1,113 @@
+require 'open3'
+require 'timeout'
+
+# Starts the Sauce Connect Proxy
+
+module Cdo
+  module SauceConnect
+    # How many seconds to wait for the "you may start your tests" message
+    SC_START_TIMEOUT_S = 120
+    SC_START_MESSAGE = "you may start your tests"
+    SC_STDOUT_PREFIX = "Sauce Connect Proxy"
+
+    class << self
+      # Starts the Sauce Connect Proxy, which allows tunneling connections from our local server to Sauce Labs
+      # This method blocks until sc prints the "you may start your tests" message to log/sc.log
+      #
+      # @param [Boolean] daemonize - if true, sc will continue running in the bg even when ruby exits
```

Comment:
What's the intended way for a developer to stop sc when running with daemonize enabled?

Reply:
Short answer: `ps` in the local terminal and kill PID, or what I do is `killall sc`.

I hemmed and hawed on this one. I was very tempted to make `bin/sauce_connect_stop`, but there's no effective way to implement this without either doing something I didn't feel safe "hiding" (=wrapper for `killall sc`), or without going absolutely bananas and basically implemented what sc should have done in the first place (a proper monitoring/control process that tracks pids etc and has proper start/stop/status control).

Its crazy to me that the sauce connect team didn't even implement "wait to daemonize until tunnel is started" mode on their size.

I have some emo about the sauce connect team today lol. v5 felt like a terrible hackeneyed rewrite with lots of pointless breakage.

Reply:
Maybe I'll add a note to the "started, success" message that one could kill it with `killall sc`. That way its visible that you're doing a wildcard kill, and folks can decide for themselves if they want to use six-shooters or a more measured approach 🤠

Reply:
OK, here's a quick fix, I'll print the suggested stop command (`killall sc`, which is what I actually use instead of `kill $PID`, because once its daemonized, you pretty much run `bin/dashboard-server`, and there's no way I'm looking back each time to find the pid again).

We print the pid, so if somebody wants to be more precise, they can easily get at it, and anyone that precise knows how to do a kill command anyway ;-)

Reply:
Output looks like this now:
```
> bin/sauce_connect
Sauce Connect Proxy: starting sc, pid = 90896, see log at /Users/seth/src/code-dot-org/log/sc.log
Sauce Connect Proxy: SUCCESS, sc is running in the background, you can stop it with `killall sc`
>
```

Reply:
Oh wow, that's frustrating. Daemons are frustrating even when they do have decent support tooling, much less when we have to manually `kill` things.

But your solution looks great to me! Love the idea of a quick & dirty default recommendation combined with surfacing the information required for something more tactical; feels like we're getting the best of both worlds